src/packages/sdk/src/utils/loadEnv.ts

```diff
@@ -16,97 +17,63 @@ type CLIArgs =
       '--telemetry-information': StringConstructor
     }>
 
-interface DotenvResult {
-  error?: Error
-  parsed?: {
-    [name: string]: string
-  }
-}
+type DotenvResult = dotenv.DotenvConfigOutput & {
+  ignoreProcessEnv?: boolean | undefined;
+};
 
 interface LoadEnvResult {
   message: string
-  dotenvResult: DotenvResult
+  path: string
+  dotenvResult:  DotenvResult
 }
-
+/**
+ * Tries load env variables
+ *  1. Load .env from project root
+ *  1. Load first .env from possible schema locations and throw if there are env clashes with root .env
+ *    1. Read location from schema arg --schema
+ *    1. Read location from pkgJSON "prisma": {"schema": "/path/to/schema.prisma"}
+ *    1. Read from default location ./prisma/.env
+ */
 export function tryLoadEnv(
   args: CLIArgs,
   opts: { cwd: string } = { cwd: process.cwd() },
 ): void {
-  let envInfo: LoadEnvResult | null = null
-  let schemaPathFromPkgJson: string | null = null
-
-  // 1 -Check --schema directory
-  if (args && args['--schema']) {
-    envInfo = tryLoadEnvFromSchemaArgs(args['--schema'])
-  }
-  // 2 - Check package.json for `prisma.schema` configuration
-  else if ((schemaPathFromPkgJson = readSchemaPathFromPkgJson()) !== null) {
-    envInfo = tryLoadEnvFromPkgJson(schemaPathFromPkgJson, opts.cwd)
+  const rootEnvInfo = loadEnvFromProjectRoot(opts)
+  const schemaEnvPathFromArgs = schemaPathToEnvPath(args['--schema'])
+  const schemaEnvPathFromPkgJson = schemaPathToEnvPath(readSchemaPathFromPkgJson())
+
+  const schemaEnvPaths = [
+    schemaEnvPathFromArgs,        // 1 - Check --schema directory for .env
+    schemaEnvPathFromPkgJson,     // 2 - Check package.json schema directory for .env
+    './prisma/.env',              // 3 - Check ./prisma directory for .env
+    './.env'                      // 4 - Check cwd for .env
```

Comment:
I might have missed something, but for 4, I think we'd want to check the same dir as where your package.json resides, rather than cwd.

Reply:
The logic for loading the env var from your project root would cover that case.

Reply:
https://github.com/prisma/prisma/pull/3971/files/f3af22fcacc7827d1c35688691d3c730f9c0ddfe#diff-59890438ec39361cbc9c8bf8a8772c311453c15035dd4e82351b04d70980fcf4R41